pytradfri/resource.py

```diff
@@ -5,39 +5,102 @@
 from datetime import datetime
 from typing import Any, Callable, Dict, List, Union
 
+from pydantic import BaseModel, Field
+
+from pytradfri.device.signal_repeater import SignalRepeaterResponse
+
 from .command import Command, TypeProcessResultCb
-from .const import ATTR_CREATED_AT, ATTR_ID, ATTR_NAME
-from .typing import ApiResourceResponse
+from .const import (
+    ATTR_APPLICATION_TYPE,
+    ATTR_CREATED_AT,
+    ATTR_DEVICE_BATTERY,
+    ATTR_DEVICE_FIRMWARE_VERSION,
+    ATTR_DEVICE_INFO,
+    ATTR_DEVICE_MANUFACTURER,
+    ATTR_DEVICE_MODEL_NUMBER,
+    ATTR_DEVICE_POWER_SOURCE,
+    ATTR_DEVICE_SERIAL,
+    ATTR_ID,
+    ATTR_LAST_SEEN,
+    ATTR_LIGHT_CONTROL,
+    ATTR_NAME,
+    ATTR_OTA_UPDATE_STATE,
+    ATTR_REACHABLE_STATE,
+    ATTR_START_BLINDS,
+    ATTR_SWITCH_PLUG,
+    ROOT_AIR_PURIFIER,
+    ROOT_SIGNAL_REPEATER,
+)
+from .device.air_purifier import AirPurifierResponse
+from .device.blind import BlindResponse
+from .device.light import LightResponse
+from .device.socket import SocketResponse
 
 # type alias
 TypeRaw = Dict[str, Union[str, int, List[Dict[str, Union[str, int]]]]]
 TypeRawSimple = Dict[str, Union[str, int]]
 TypeRawList = Dict[str, List[Dict[str, Union[str, int]]]]
 
 
+class DeviceInfoResponse(BaseModel):
```

Comment:
We want to move this to the device.py module but we can do that in a follow up PR to avoid having to move around the names inside device.py.